src/main/asciidoc/client-applications.adoc

```diff
@@ -388,16 +388,16 @@ Detailed descriptions of connection pool settings available via driver configura
 
 `MaxConnectionLifetime`::
 Pooled connections older than this threshold will be closed and removed from the pool.
-Such removal happens during connection acquisition so that new session is never backed by an old connection.
+The actual removal happens during connection acquisition so that the new session returned is never backed by an old connection.
 Setting this option to a low value will cause a high connection churn and might result in a performance drop.
-It is recommended to set driver's maximum lifetime to a value smaller than the maximum lifetime configured in its application system infrastructure (such as operation system, router, load balancer, proxy and firewall).
+It is recommended to pick a value smaller than the maximum lifetime exposed by the surrounding system infrastructure (such as operating system, router, load balancer, proxy and firewall).
 Negative values result in lifetime not being checked.
 Default value: 1h.
 
 `MaxConnectionPoolSize`::
-This setting defines the maximum total number of connections allowed to be managed by the connection pool on each host.
-In other words, for a direct driver, this sets the maximum amount of connections towards a single database.
-For a routing driver this sets the maximum amount of connections towards each cluster member.
+This setting defines the maximum total number of connections allowed to be managed by the connection pool per host.
```

Comment:
Proposed wording to make the sentence more clear:
This setting defines the maximum total number of connections allowed, per host, to be managed by the connection pool.
Please verify that the meaning is still correct. Thanks!